New version 2.1

SimonsUnikaNamn opened the pull request:
La till ett sätt att skapa egna ConnectionAmount med validering på inputen så att man slipper skapa en massa nya statiska exempel när man vill ha nya antal i poolen

Commit:
Made it possible to add your own connection pool settings

runeflobakk commented on src/main/java/no/digipost/http/client/Validation.java:
```diff
@@ -0,0 +1,26 @@
+ */
+package no.digipost.http.client;
+
+public class Validation {
```
👍 
Details på sånn util-klasser: lag de `final`, og legg til en `private` constructor.

runeflobakk commented on src/main/java/no/digipost/http/client/Validation.java:
```diff
@@ -0,0 +1,26 @@
+
+public class Validation {
+
+    public static int NOrGreater(int value, int threshold, String nameOfValue) {
```
Metodenavnet bør begynne med lowercase. Foreslår `nOrGreater(..)`, men foretrekker egentlig `equalOrGreater(..)` da jeg synes "equal" kommuniserer intent litt bedre enn "n".

runeflobakk commented on src/main/java/no/digipost/http/client/Validation.java:
```diff
@@ -0,0 +1,26 @@
+public class Validation {
+
+    public static int NOrGreater(int value, int threshold, String nameOfValue) {
+        if (value < threshold) {
```
Sjekker bare "større enn", godtar ikke dersom `value == threshold` selv om metoden sier den skal gjøre det.
`DigipostHttpClientMillisecondTimeouts` godtar nå ikke 0-verdi (uendelig) som timeout, noe den bør gjøre for debugging.

runeflobakk commented on src/main/java/no/digipost/http/client/DigipostHttpClientDefaults.java:
```diff
@@ -128,7 +127,9 @@
         public final int maxPerRoute;
 
-        private ConnectionAmount(int maxTotal, int maxPerRoute) {
+        public ConnectionAmount(int maxTotal, int maxPerRoute) {
```
Synes `ConnectionAmount` bør følge samme pattern som `DigipostHttpClientMillisecondTimeouts`. Altså bør `ConnectionAmount` extractes som egen klasse -> `DigipostHttpClientConnectionAmount`, og ha litt metoder for å endre verdiene. `DigipostHttpClientMillisecondTimeouts` har også en metode `isPotentiallyDangerous`, som kalles i `DigipostHttpClientFactory`. Bør `DigipostHttpClientConnectionAmount` også ha en slik? Er det noen verdier som bør være gyldige, men som kun bør brukes f.eks. test?

Jeg foreslår at vi parprogrammerer disse endringene :)

@froden nevnte at han synes det var litt forvirrende at for å lage custom `DigipostHttpClientMillisecondTimeouts` måtte han begynne med `DigipostHttpClientDefaults.DEFAULT_TIMEOUTS_MS` og "modifisere" denne, men jeg foreslår å fikse dette i en egen pull-request, og at vi nå følger "patternet" som allerede er etablert her.

runeflobakk reviewed (changes requested):
Foreslår å fjerne disse commitene og rebase:
- Revert "Release:prepare for next development iteration" 58a726e
- Revert "Release:prepare release 2.1" 557c25f

Dette er ordnet i 2a58287 i `master`

Commit:
final class Validation

Gjør Validation-klassen final og package private, skal ikke brukes utenfor biblioteket.

Commit:
Ignore *.iml files

Commit:
Connection pool føljer pattern

ConnectionAmount har skiftats till DigipostHttpClientConnectionAmount och føljer standarden som fanns i DigipostHttpClientMillisecondsTimeout

Commit:
Added a 3 to class path

The 3 is added so that the new 3.0 version can co-exist with older version since this version contains breaking changes

Commit:
Endrade artifact id

Artifact id är nu http-client-builder3 för att reflektera den nya classpathen

Commit:
Release:prepare release 3.0

Commit:
Release:prepare for next development iteration
